### DISCUSSION TITLE
Can the client do an upload only where it archives data to OneDrive?

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#1926] [Q&A] [answered]
I use deja dup for backups, which is an incremental backup tool for personal files.

Could I use the client to 'push' the backup into the cloud, i.e. not actually have a copy of the backup sitting on my drive taking up space, just having a full copy of the backup encrypted in the onedrive cloud?

If so, that would be amazzzzing. :)

n.b. I also have other personal files I'd like to do this with... almost like I am using OneDrive as a sort of drive in its own right rather than a sync... so, you can see the files on nemo/file explorer locally, but they are actually sitting on OneDrive only, and only accessed on demand if you want them (e.g. open a word doc, view an image, whatever).

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 

> I use deja dup for backups, which is an incremental backup tool for personal files.
> 
> Could I use the client to 'push' the backup into the cloud, i.e. not actually have a copy of the backup sitting on my drive taking up space, just having a full copy of the backup encrypted in the onedrive cloud?
> 
> If so, that would be amazzzzing. :)

Please read the usage document - https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-one-way-upload-sync


> n.b. I also have other personal files I'd like to do this with... almost like I am using OneDrive as a sort of drive in its own right rather than a sync... so, you can see the files on nemo/file explorer locally, but they are actually sitting on OneDrive only, and only accessed on demand if you want them (e.g. open a word doc, view an image, whatever).

Please read https://github.com/abraunegg/onedrive/issues/757#issuecomment-768532497

**Comment by johnsidoli** [comment]
Ah yes, thanks! I'm guessing if you just wanted this for one folder (e.g. "/archive", you'd need to setup a new config, right?

**Reply by abraunegg** [reply]
@johnsidoli 
If this is in relation to your upload only query .. then yes ... but please read the documentation and ensure that it is also *excluded* from any other configurations you are using.

**Reply by johnsidoli** [reply]
Thanks, I've got it and understood.

However, I see that the functionality is more like the upload only, but not upload only and access?

My original query was more if it could do something like this: https://tresorit.com/features/tresorit-drive

Where you can access on the fly, without download.

Perhaps that's a way down the road yet :)

**Reply by abraunegg** [reply]
@johnsidoli 
> Where you can access on the fly, without download.

This is known as 'on-demand' functionality.  Please read https://github.com/abraunegg/onedrive/issues/757#issuecomment-768532497 more carefully.
